npm/deploy/Options.kt

```diff
@@ -0,0 +1,72 @@
+package com.vaticle.bazel.distribution.npm.deploy
+
+import com.vaticle.bazel.distribution.npm.deploy.Options.CommandLineParams.NPM_PATH
+import com.vaticle.bazel.distribution.npm.deploy.Options.CommandLineParams.RELEASE_REPO
+import com.vaticle.bazel.distribution.npm.deploy.Options.CommandLineParams.SNAPSHOT_REPO
+import com.vaticle.bazel.distribution.npm.deploy.Options.Env.DEPLOY_NPM_TOKEN
+import com.vaticle.bazel.distribution.npm.deploy.Options.RepositoryType.RELEASE
+import com.vaticle.bazel.distribution.npm.deploy.Options.RepositoryType.SNAPSHOT
+import picocli.CommandLine
+
+class Options {
```

Comment:
Options are constructed from command line arguments and environment variables.

We use picocli for command line parsing. Argument names are defined in the inner object `Options.CommandLineParams`. Env var names are defined in `Options.Env`.

Parameters after a `--` in the command are passed in to the `params` variable (because of its annotation `@CommandLine.Parameters`). For example, in `java -jar DeployNPM.jar --npm_path /usr/bin/node -- snapshot`, `params` will have a value of `["snapshot"]`.

We enumerate, and validate, these positional parameters in `Options.RepositoryType`, ensuring that the user may only pass in a single positional parameter; this parameter represents the _repo type_; and the repo type must be "snapshot" or "release".